exercises/bracket-push/example.rs

```diff
@@ -0,0 +1,62 @@
+// The code below is a stub. Just enough to satisfy the compiler.
+// In order to pass the tests you can add-to or change any of this code.
+
+use std::collections::HashMap;
+pub struct Brackets {
+    all: Vec<char>,
+    pairs: HashMap<char, char>,
+}
+
+impl<'a> From<&'a str> for Brackets {
+    fn from(i: &str) -> Self {
+        Brackets::new(String::from(i), None)
+    }
+}
+
+impl Brackets {
+    pub fn new(s: String, pairs: Option<HashMap<char, char>>) -> Self {
+        let p = match pairs {
+            Some(x) => x,
+            None => Brackets::default(),
+        };
+
+        let valid = p.keys().chain(p.values()).collect::<Vec<_>>();
+        let all = s.chars().filter(|c| valid.contains(&c)).collect::<Vec<char>>();
+
+        Brackets {
+            all: all,
+            pairs: p.clone(),
+        }
+    }
+
+    fn default() -> HashMap<char, char> {
+        let mut d = HashMap::new();
+        d.insert('[', ']');
```

Comment:
It's times like these when I wish for some hash map literals. (This is possible today with macros, if one really wishes it, though doesn't seem useful enough for this example implementation)

And maybe a way to share this default against all instances of `Brackets` that get the default map, but this is tricky.

No action needed, only musing about language capabilities.